dproc/dproc.go

```diff
@@ -0,0 +1,166 @@
+	srcDb := sourceProfile.Config.ShardConfigurationDataproc.SchemaSource.DbName
+	spDb := targetProfile.Conn.Sp.Dbname
+	spInstance := targetProfile.Conn.Sp.Instance
+	spProject := targetProfile.Conn.Sp.Project
+	if len(sourceProfile.Config.ShardConfigurationDataproc.DataprocConfig.Hostname) > 1 {
+		host = sourceProfile.Config.ShardConfigurationDataproc.DataprocConfig.Hostname
+	}
+	if len(sourceProfile.Config.ShardConfigurationDataproc.DataprocConfig.Port) > 1 {
+		port = sourceProfile.Config.ShardConfigurationDataproc.DataprocConfig.Port
+	}
+
+	jdbcParams := map[string]string{}
+	if sourceProfile.Driver == constants.MYSQL {
+		jdbcParams["url"] = fmt.Sprintf("jdbc:mysql://%s:%s/%s?user=%s&password=%s", host, port, srcDb, user, pwd)
+		jdbcParams["driver"] = "com.mysql.jdbc.Driver"
+		jdbcParams["sql"] = fmt.Sprintf("select * from %s.%s", srcSchema, srcTable)
+		jdbcParams["jar"] = "gs://dataproc-templates/jars/mysql-connector-java.jar"
+	} else {
+		return DataprocRequestParams{}, fmt.Errorf("dataproc migration for source %s not supported", sourceProfile.Driver)
+	}
+
+	jarFileUris := []string{"file:///usr/lib/spark/external/spark-avro.jar",
+		"gs://dataproc-templates-binaries/latest/java/dataproc-templates.jar",
+		jdbcParams["jar"]}
+
+	args := []string{"--template",
+		"JDBCTOSPANNER",
+		"--templateProperty",
+		"project.id=" + spProject,
+		"--templateProperty",
+		"jdbctospanner.jdbc.url=" + jdbcParams["url"],
+		"--templateProperty",
+		"jdbctospanner.jdbc.driver.class.name=" + jdbcParams["driver"],
+		"--templateProperty",
+		"jdbctospanner.sql=" + jdbcParams["sql"],
+		"--templateProperty",
+		"jdbctospanner.output.instance=" + spInstance,
+		"--templateProperty",
+		"jdbctospanner.output.database=" + spDb,
+		"--templateProperty",
+		"jdbctospanner.output.table=" + srcTable,
+		"--templateProperty",
+		"jdbctospanner.output.primaryKey=" + primaryKeys,
+		"--templateProperty",
+		"jdbctospanner.output.saveMode=" + saveMode,
+		"--templateProperty",
+		"jdbctospanner.output.batch.size=" + batchSize,
+		"--templateProperty",
+		"jdbctospanner.jdbc.fetchsize=" + fetchSize}
+
+	dataprocRequestParams := DataprocRequestParams{
+		Project:       spProject,
+		TemplateArgs:  args,
+		JarFileUris:   jarFileUris,
+		SubnetworkUri: subnet,
+		Location:      location,
+	}
+
+	return dataprocRequestParams, nil
+}
+
+func CreateDataprocBatchClient(location string) (*dataproc.BatchControllerClient, error) {
+	ctx := context.Background()
+	batchEndpoint := fmt.Sprintf("%s-dataproc.googleapis.com:443", location)
+	batchClient, err := dataproc.NewBatchControllerClient(ctx, option.WithEndpoint(batchEndpoint))
+	return batchClient, err
+}
+
+func TriggerDataprocTemplate(batchClient *dataproc.BatchControllerClient, srcTable string, srcSchema string, primaryKeys string, dataprocRequestParams DataprocRequestParams) (string, error) {
+	ctx := context.Background()
+
+	fmt.Printf("Triggering Dataproc template for %s.%s\n", srcSchema, srcTable)
```

Comment:
please use the logger here

Reply:
if it would make sense the log the non-sensitive parameters of the job here

Reply:
Done, 
Commit: https://github.com/cloudspannerecosystem/harbourbridge/pull/520/commits/a245b849a66861ec60e6d44c3294e6e052f233bb